### DISCUSSION TITLE
Alert if worker node is down

### DISCUSSION THREAD

**Discussion by bb3454** [jhuckaby/Cronicle#703] [Q&A] [answered]
I want to be able to monitor if worker nodes are up/down. I see entries in the primary node activity log if connectivity is lost but haven't seen anything to alert me if one is down. Is there an API which can give status of a worker node, similar to a ping? Or is there a different way to achieve this?

**Comment by jhuckaby** [comment, marked as answer]
There's an undocumented API you can use.  See this discussion here: https://github.com/jhuckaby/Cronicle/issues/674

First, you need to create an administrative API key.  That process is described here: https://github.com/jhuckaby/Cronicle/discussions/479

Then you can hit the new `get_servers` API, which will tell you which servers are up/down.  Look for the `disabled` property inside each server object.  It'll be set to `true` if the server is down. 
 Example:

```
http://mini.local:3012/api/app/get_servers?api_key=da323d6cb9c29a37a59bf95d1726920b&pretty=1
```

```json
{
	"code": 0,
	"servers": {
		"mini.local": {
			"hostname": "mini.local",
			"ip": "192.168.3.46",
			"master": 1,
			"uptime": 95,
			"data": {
				"cpu": 5.7,
				"mem": 57663488
			},
			"disabled": 0
		},
		"joework.local": {
			"hostname": "joework.local",
			"ip": "192.168.3.40",
			"master": 0,
			"uptime": 0,
			"data": {},
			"disabled": true
		}
	}
}
```

Notice the `"disabled": true` on the `joework.local` server.

Please note that it takes up to a minute to "realize" that a server has gone offline.

**Reply by jhuckaby** [reply]
Note that Cronicle v2 will have a fully supported and official way to do this, by way of web hooks.  You will be able to register a web hook for any activity, including a server disconnection event.

**Reply by bb3454** [reply]
Thanks this should give me what I want for now. Any timelines on v2?

**Reply by jhuckaby** [reply]
Early 2024, with any luck!
